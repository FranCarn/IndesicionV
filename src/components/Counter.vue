<template>
  <h2>{{ customTitle }}</h2>
  <p>{{ counter }}</p>
  <button @click="decrement">- 1</button>
  <button @click="increment">+ 1</button>

  <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
</template>
<script>
export default {
  name: "CounterVue",
  props: {
    header: String,
    start: {
      type: Number,
      default: 100,
      required: true,
      validator(value) {
        return value >= 0;
      },
    },
  },
  data() {
    return {
      counter: this.start || 0,
    };
  },
  methods: {
    increment() {
      this.counter = this.counter + 1;
    },
    decrement() {
      if (!this.counter) return;
      this.counter = this.counter - 1;
    },
  },
  computed: {
    squareCounter() {
      return this.counter * this.counter;
    },
    customTitle() {
      return this.header || "Counter";
    },
  },
};
</script>

<style scoped>
button {
  background-color: #64b687;
  border-radius: 5px;
  border: 1px solid #fff;
  cursor: pointer;
  margin: 0 5px;
  padding: 5px 15px;
  transition: 0.3s ease-in-out;
}
button:hover {
  background-color: #5aa67b;
  transition: 0.5s ease-in-out;
}
</style>
