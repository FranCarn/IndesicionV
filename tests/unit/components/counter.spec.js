import { shallowMount, mount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";

describe("Counter.vue Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Counter, { props: { start: 100 } });
  });

  afterEach(() => {
    wrapper = null;
  });

  test("should match with snapshop", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("should checking h2 default value", () => {
    expect(wrapper.find("h2").exists()).toBeTruthy();
    const h2 = wrapper.find("h2").text();
    expect(h2).toBe("Counter");
  });
  test("should show default value 100 in paragraph", () => {
    // const p = wrapper.findAll("p");
    // expect(p.length).toBeGreaterThanOrEqual(2);
    // expect(p[0].text()).toBe("100");
    const p = wrapper.find('[data-testid="counter"]').text();
    expect(p).toBe("100");
  });
  it("should increase & decrease value counter", async () => {
    const [decreaseBtn, increaseBtn] = wrapper.findAll("button");
    await decreaseBtn.trigger("click");
    await increaseBtn.trigger("click");
    await increaseBtn.trigger("click");

    const counter = wrapper.find('[data-testid="counter"]').text();
    expect(counter).toBe("101");
  });
  test("should check value in props", () => {
    // const value = wrapper.props('start');
    const { start } = wrapper.props();
    const value = wrapper.find('[data-testid="counter"]').text();
    expect(Number(value)).toBe(start);
  });
  it("should show title prop", () => {
    const props = {
      start: 100,
      header: "Hi world",
    };
    const wrapper = shallowMount(Counter, { props });
    const value = wrapper.find('[data-testid="counter"]').text();
    const text = wrapper.find("h2").text();
    expect(Number(value)).toBe(props.start);
    expect(text).toBe(props.header);
  });
});
