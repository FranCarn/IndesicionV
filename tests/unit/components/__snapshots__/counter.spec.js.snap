// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue Component should match with snapshop 1`] = `
<h2>Counter</h2>
<p data-testid="counter">100</p>
<button>- 1</button>
<button>+ 1</button>
<p>100 <sup>2</sup> = 10000</p>
`;
